ProcessQueue/__init__.py

```diff
@@ -159,320 +190,343 @@ def main(msg: func.QueueMessage):
-        data.append(scan_report_table_entry)
+        table_entries_to_post.append(table_entry)
 
-    # Create JSON array
-    json_data = json.dumps(data)
+    print("POST tables", datetime.utcnow().strftime("%H:%M:%S.%fZ"))
     # POST request to scanreporttables
-    response = requests.post(
-        "{}scanreporttables/".format(api_url), data=json_data, headers=headers
+    tables_response = requests.post(
+        "{}scanreporttables/".format(api_url), 
+        data=json.dumps(table_entries_to_post), 
+        headers=headers
     )
-    print("TABLE SAVE STATUS >>>", response.status_code)
+    print("POST tables finished", datetime.utcnow().strftime("%H:%M:%S.%fZ"))
+
+    print("TABLE SAVE STATUS >>>", tables_response.status_code)
+    # Error on failure
+    if tables_response.status_code != 201:
+        raise HTTPError(' '.join(['Error in table save:', str(tables_response.status_code), str(json.dumps(table_entries_to_post))]))
 
     # Load the result of the post request,
-    response = json.loads(response.content.decode("utf-8"))
+    tables_content = json.loads(tables_response.content.decode("utf-8"))
 
     # Save the table ids that were generated from the POST method
-    for element in range(len(response)):
-        table_ids.append(response[element]["id"])
-    # print("TABLE IDs", table_ids)
+    table_ids = [element["id"] for element in tables_content]
+
+    print("TABLE IDs", table_ids)
 
     """
     POST fields per table:
     For each row in Field Overview create an entry for scan_report_field,
     Empty row signifies end of fields in a table
-    Append field entry to data[] list,
+    Append field entry to field_entries_to_post[] list,
     Create JSON array with all the field entries, 
     Send POST request to API with JSON as input,
     Save the response data(field ids,field names) in a dictionary
     Set the current working sheet to be the same as the current table
     Post the values for that table
     """
-    idx = 0
-    data = []
+    table_idx = 0
+    field_entries_to_post = []
     names_x_ids = {}
+
+    # Loop over all rows in Field Overview sheet.
     # For sheets past the first two in the Scan Report
     # i.e. all 'data' sheets that are not Field Overview and Table Overview
     worksheet_idx = 2
+    print("Start fields loop", datetime.utcnow().strftime("%H:%M:%S.%fZ"))
 
     for i, row_cell in enumerate(
-        ws.iter_rows(min_row=2, max_row=ws.max_row + 1), start=2
+        fo_ws.iter_rows(min_row=2, max_row=fo_ws.max_row + 1), start=2
     ):
 
-        if idx >= len(table_ids):
+        if table_idx >= len(table_ids):
             continue
         # If fraction empty or fraction unique is empty set to 0(decimal)
-        if not (ws.cell(row=i, column=8).value):
-            ws.cell(row=i, column=8).value = 0.0
-        if not (ws.cell(row=i, column=10).value):
-            ws.cell(row=i, column=10).value = 0.0
+        if not (fo_ws.cell(row=i, column=8).value):

```

Comment:
Dont really get all these lines. Earlier there was a change to iterate over the cells, and then `row_cell.value` was used. I guess this is because a different column is being grabbed? 

Not sure if my comments makes sense or not. Just double checking there's not a mistake or a more readable way of doing this. Im sure it's fine as the previous code worked.

Reply:
I've written a helper function `default_zero()` to neaten this up - I feel like there's possibly an easier way, but I can't see it right now.